accepted/2020/mono-convergence/globalization-support-in-.net5.md

```diff
@@ -0,0 +1,83 @@
+
+## .NET Globalization Invariant Mode
+There is already support for the globalization [invariant mode](https://github.com/dotnet/runtime/blob/master/docs/design/features/globalization-invariant-mode.md) since .NET Core 2.0. This will be the most size effective globalization mode for .NET5 which can be set by the developers. We are going to use existing logic and tweak it where necessary to allow IL Linker to remove any globalization code from the final app when running in invariant mode.
+
+## Size Sensitive Platforms
+Existing Xamarin platforms allow customers to choose which set of collation data they would like to add to their apps if any. This still makes sense for .NET5 as on some of the platform ICU dependency is not available from the system and we’ll need to bundle it to the final app. We could make this available to everyone but it’s most likely we will focus only on platforms which are size sensitive in .NET5 time frame.
+
+### Android
+Android has historically relied on ICU for some kind of globalization support.  From basically the beginning, `Locale` , `Character`, and some classes in `java.text` were provided in the platform globalization.  If you needed more than that, you could access `libicuuc` directly or choose to bundle another version in your APK.  That was pretty much the story up until API level 23.
+
+In API level 24 and higher, the platform white lists its version of `libicuuc` and also includes a subset of [ICU4J](https://developer.android.com/guide/topics/resources/internationalization#relation) API’s.  It’s important to note that [ICU4C](https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/) (accessing `libicuuc` directly) is not part of the Android NDK.  What this means is that OEM’s are not required to follow any type of rules and they could limit / alter the data files as they see fit.  The potential for undefined behavior and additional testing on our part is very real. 
+
+Consult this [issue](https://github.com/android/ndk/issues/548#issuecomment-395561629) for detailed commentary on the challenges of making ICU an NDK API.
+
+### What Does This Mean For Us?
+The good news is that p-invoking from managed code works.  Assuming `libicuuc` continues to be white listed, this should not be a problem.  It would be wise if we explored using the `ICU4J`  API subset that’s exposed in the event they decide to tighten things up.  
+
+To summarize:
+
+* We can p-invoke into the system `libicuuc`.  Globalization for .net5 should work normally
+* Android 4.3 (API level 18) started bundling ICU 50 and our minimum version requirement is satisfied. 
+* OEM’s can potentially make life difficult by altering locales / globalization features
+* We should explore trying to interop with the `ICU4J` API and determine what the implications, if any are.
+
+### iOS/tvOS
```

Comment:
That section should also include **watchOS**. 

watchOS it has application size limitations (50MB) which makes brining additional dependencies very hard.

Reply:
It's my understanding that watchOS is not included in .net5.  Would it make sense to consider now?

Reply:
care to link me to that decision ? because it's right there in https://devblogs.microsoft.com/dotnet/introducing-net-5/

Reply:
@steveisok (and @marek-safar) - Can we please get confirmation on what platforms / archs / features are "not included in .net 5" as of right now. 

I'd like to get a handle on it, and be able to get it in front of customers before release this fall.

Reply:
For some reason I'm not getting notifications, so my apologies.  I had thought that was official, but that's clearly my assumption may be incorrect.  @marek-safar ?

Reply:
@steveisok and @marek-safar  Did you get any clarification on WatchOS support?  I'm building a new app using Xamarin and WatchOS and I don't want to waste my time if support will be dropped.  

I have two apps in the store ([here](https://apps.apple.com/us/app/voice-in-a-can/id1339696720?platform=appleWatch) and [here](https://apps.apple.com/us/app/kanbann/id1480933645#?platform=appleWatch)) currently using Xamarin and WatchOS which would be orphaned if you drop support.

Reply:
@DamianMehers Xamarin won't drop WatchOS support. Here we are discussing the priorities of how to migrate that to .NET 5/6/7.

Reply:
Understood - my concern is that if it doesn't make it to .NET 5 then I assume eventually support will disappear.